guardian/newsletters-nx
# Dblatcher/s3 env vars

## What does this change?

Removes the hard-coded s3 configuration, has the storage service read the s3 params from environment variables instead. 

The application will use an `S3DraftStorage` if all three required s3 params are present in the environment variables, unless the USE_LOCAL_STORAGE variable is true

gitignores .env.local files at any path.

## How to test

Locally:
 - Copy and rename **apps/newsletters-api/env.local.example.txt** to  **apps/newsletters-api/.env.local**
 - update the S3_BUCKET value in .env.local to then name of an s3 bucket in the "developerPlayground" account

run locally, `npm run dev` and open http://localhost:3000/api/drafts - if you don't have developerPlayground from [janus](https://janus.gutools.co.uk/), the request should fail with a 403 error. If you get credentials (and restart the application), it should list the draft files in the bucket (if any).

Setting USE_LOCAL_STORAGE=true in your **.env.local** (and restart the application). [The API page](http://localhost:3000/api/drafts) will show the 5x sample drafts hard coded in the `makeInMemoryStorageInstance` function.

## How can we measure success?

Moves us closer to a "real world" implementation that could work in PROD. No hard coded S3 parameters.

## Have we considered potential risks?

We should still request help from DevX to adapt this for production - not sure how best to obtain s3 credentials on CODE and PROD (use parameters store / secrets manager?).
